### ISSUE TITLE
VariableInitialization false positive for custom managed record with Initialize operator

### ISSUE THREAD

**Issue by SebastianZieglerPOI** [integrated-application-development/sonar-delphi#468]
### Prerequisites

- [x] This improvement has not already been suggested.
- [x] This improvement should not be implemented as a separate rule.

### Rule to improve

Variables must be initialized before being used

### Improvement description

The `VariableInitialization` rule reports "Initialize this variable before
using it" when a local custom managed record is passed to a routine, although
the record has an `Initialize` operator that assigns all its fields.

The warning also persists when the record declares and implements a `Finalize`
operator. In the observed case, two fields are changed after the local record
is created, and the warning appears when the record is passed as an argument.

Illustrative minimal example (using fictional names):

```pascal
program ManagedRecordExample;

{$APPTYPE CONSOLE}

type
  TRequestOptions = record
    Code: Integer;
    Name: string;
    class operator Initialize(out Dest: TRequestOptions);
    class operator Finalize(var Dest: TRequestOptions);
  end;

class operator TRequestOptions.Initialize(out Dest: TRequestOptions);
begin
  Dest.Code := 0;
  Dest.Name := '';
end;

class operator TRequestOptions.Finalize(var Dest: TRequestOptions);
begin
  Dest.Name := '';
end;

procedure ProcessRequest(const Options: TRequestOptions);
begin
  WriteLn(Options.Code, ' ', Options.Name);
end;

procedure Run;
var
  Options: TRequestOptions;
begin
  Options.Name := 'Example';
  ProcessRequest(Options); // Reported as uninitialized in the observed pattern
end;

begin
  Run;
end.
```

The rule should account for Delphi's automatic invocation of `Initialize`
when assessing the subsequent use of a local custom managed record.

### Rationale

A local custom managed record is initialized automatically by Delphi through
its `Initialize` operator. An explicit assignment solely to satisfy the rule
is therefore unnecessary. Implementing `Finalize` does not change the
initialization behavior or remove the warning.

This is a false positive in `VariableInitialization`, not a parsing problem
or the implicit-Self syntax issue addressed in #416. Ordinary records without
an `Initialize` operator should retain the rule's existing checks.

Documentation:
https://docwiki.embarcadero.com/RADStudio/Florence/en/Custom_Managed_Records
